packages/alfa-rules/src/common/predicate/is-rendered.ts

```diff
@@ -3,8 +3,18 @@ import { Element, Comment, Node } from "@siteimprove/alfa-dom";
 import { Predicate } from "@siteimprove/alfa-predicate";
 import { Style } from "@siteimprove/alfa-style";
 
+import { hasName } from "@siteimprove/alfa-dom/src/node/element/predicate/has-name";
+const { and } = Predicate;
+
 export function isRendered(device: Device): Predicate<Node> {
   return (node) => {
+    // descendants of iframe are not rendered, they are fallback content for legacy browsers…
+    // @see https://html.spec.whatwg.org/multipage/iframe-embed-object.html#the-iframe-element
+
+    if (node.ancestors().some(and(Element.isElement, hasName("iframe")))) {
+      return false;
+    }
```

Comment:
This will cause `isRendered()` to walk to the root of every node visited, making the complexity quadratic in the depth of the node for which the function was called. As `isRendered()` already traverses the ancestor chain, it's enough to check if the _parent_ is an `<iframe>` element.